samples/ar-view-node/src/main/java/io/github/sceneview/sample/arviewnode/MainActivityViewMvcImpl.kt

```diff
@@ -0,0 +1,133 @@
+package io.github.sceneview.sample.arviewnode
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import io.github.sceneview.ar.ARSceneView
+import io.github.sceneview.sample.arviewnode.databinding.ActivityMainBinding
+import java.util.Collections
+
+class MainActivityViewMvcImpl(
+    private val context: Context,
+    inflater: LayoutInflater,
+    parent: ViewGroup?,
+) : View.OnClickListener, MainActivityViewMvc {
+    private val binding: ActivityMainBinding
+
+    private val _listeners: MutableSet<MainActivityViewMvc.Listener> = HashSet()
+
+    private val listeners: Set<MainActivityViewMvc.Listener>
+        get() = Collections.unmodifiableSet(_listeners)
+
+    ///////
+    // region Init
+    init {
+        binding = ActivityMainBinding
+            .inflate(
+                inflater,
+                parent,
+                false
+            )
+    }
+
+    init {
+        binding.btnAdd.setOnClickListener(this)
+        binding.btnViewNode.setOnClickListener(this)
+        binding.btnImageNode.setOnClickListener(this)
+    }
+    // endregion Init
+    ///////
+
+
+    ///////////////////////
+    // region View.OnClickListener
+    override fun onClick(v: View) {
+
+        when (v.id) {
+            binding.btnAdd.id -> {
+                listeners.forEach {
+                    it.handleOnAddClicked()
+                }
+            }
+
+            binding.btnViewNode.id -> {
+                handleOnSetViewNodeClicked()
+            }
+
+            binding.btnImageNode.id -> {
+                handleOnSetImageNodeClicked()
+            }
+        }
+    }
+
+    private fun handleOnSetViewNodeClicked() {
+        listeners.forEach {
+            it.updateActiveNodeType(MainActivity.NodeType.VIEW_NODE)
+        }
+
+        applyActiveColor(binding.btnViewNode)
+        applyDisabledColor(binding.btnImageNode)
+    }
+
+    private fun handleOnSetImageNodeClicked() {
+        listeners.forEach {
+            it.updateActiveNodeType(MainActivity.NodeType.IMAGE_NODE)
+        }
+
+        applyActiveColor(binding.btnImageNode)
+        applyDisabledColor(binding.btnViewNode)
+    }
+
+    private fun applyActiveColor(view: View) {
+        view.backgroundTintList = ContextCompat
+            .getColorStateList(
+                context,
+                R.color.main_color
+            )
+    }
+
+    private fun applyDisabledColor(view: View) {
+        view.backgroundTintList = ContextCompat
+            .getColorStateList(
+                context,
+                R.color.gray_500
+            )
+    }
```

Comment:
It's also possible to enable and disable the buttons.

Reply:
That's right, I wanted to give the buttons a specific color based on their state. 
And if I'm not wrong, click events are not triggered on a disabled Button.

Reply:
My bad, why the heck should someone click on a Button where the enabled state is false. I will fix this.

Reply:
If I use "isEnabled = true/false" the click handling is gone if set to true.

Reply:
This is strange but let's leave it as is for now :thinking:

Reply:
To avoid this, I could split the sample into two samples. One for ViewNode and one for ImageNode. That would reduce the complexity a little bit and the UI can be reduced to only one Button.